libcrux-testbench/src/mldsa.rs

```diff
@@ -0,0 +1,81 @@
+use libcrux_iot_testutil::DefmtInfoLogger;
+use libcrux_iot_testutil::*;
+extern crate alloc;
+use alloc::string::String;
+use alloc::vec;
+
+use libcrux_ml_dsa::ml_dsa_65;
+
+struct MLDSABenchState {
+    randomness_gen: [u8; 32],
+    keypair: ml_dsa_65::MLDSA65KeyPair,
+    signing_randomness: [u8; 32],
+    message: [u8; 1024],
+    signature: ml_dsa_65::MLDSA65Signature,
+}
+
+fn bench_keygen<L: EventLogger>(_l: &mut L, state: &MLDSABenchState) -> Result<(), String> {
+    let _pair = ml_dsa_65::generate_key_pair(state.randomness_gen);
+    Ok(())
+}
+
+fn bench_sign<L: EventLogger>(_l: &mut L, state: &MLDSABenchState) -> Result<(), String> {
+    let _signature = ml_dsa_65::sign(
+        &state.keypair.signing_key,
+        &state.message,
+        b"",
+        state.signing_randomness,
+    );
+    Ok(())
+}
+
+fn bench_verify<L: EventLogger>(_l: &mut L, state: &MLDSABenchState) -> Result<(), String> {
+    let _ = ml_dsa_65::verify(
+        &state.keypair.verification_key,
+        &state.message,
+        b"",
+        &state.signature,
+    );
+
+    Ok(())
+}
+
+pub fn run_benchmarks() {
+    // set up the test suite
+    let test_cases = [
+        TestCase::new("bench_keygen", bench_keygen),
+        TestCase::new("bench_sign", bench_sign),
+        TestCase::new("bench_verify", bench_verify),
+    ];
+
+    let test_suite = TestSuite::new("ML-DSA Benchmark", &test_cases);
+
+    // set up the test config
+    let test_config = TestConfig {
+        core_freq: 4_000_000,
```

Comment:
The core frequency depends on the board. Maybe make it an argument of the function and let the board calling it provider the value? Actually -- maybe let the caller provide the config? They can also parse it from the env if we do the "put config in env that gets read in build.rs" trick.

Reply:
Yeah, I don't think passing it in through env is very urgent.